src/sqlfluff/dialects/dialect_snowflake_keywords.py

```diff
@@ -49,6 +49,7 @@
 LIKE
 LOCALTIME
 LOCALTIMESTAMP
+MATCH_RECOGNIZE
```

Comment:
Had to move this to reserved keywords to stop it from being captured as an alias